File: rust/candid_parser/tests/assets/ok/malicious_doc.js
import { IDL } from '@dfinity/candid';

export const MaliciousType = IDL.Record({ 'field' : IDL.Text });

export const idlService = IDL.Service({
  'get' : IDL.Func([], [MaliciousType], ['query']),
});

export const idlInitArgs = [];

export const idlFactory = ({ IDL }) => {
  const MaliciousType = IDL.Record({ 'field' : IDL.Text });
  
  return IDL.Service({ 'get' : IDL.Func([], [MaliciousType], ['query']) });
};

export const init = ({ IDL }) => { return []; };

File: rust/candid_parser/tests/assets/ok/malicious_doc.rs
// This is an experimental feature to generate Rust binding from Candid.
// You may want to manually adjust some of the types.
#![allow(dead_code, unused_imports)]
use candid::{self, CandidType, Deserialize, Principal};
use ic_cdk::api::call::CallResult as Result;

/// */ import { malicious } from 'attacker'; console.log('injected!'); /*
/// Normal doc line
/// Another */ attempt /* to inject
#[derive(CandidType, Deserialize)]
pub struct MaliciousType {
  /// Doc comment for field with */ malicious code /* in it
  pub field: String,
}

/// Service with */ malicious */ doc
pub struct Service(pub Principal);
impl Service {
  /// Method with */ in doc comment
  pub async fn get(&self) -> Result<(MaliciousType,)> {
    ic_cdk::call(self.0, "get", ()).await
  }
}
/// Canister ID: `aaaaa-aa`
pub const CANISTER_ID : Principal = Principal::from_slice(&[]);
pub const service : Service = Service(CANISTER_ID);

